File: backend/app/schemas/chat_log.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import SessionResolution


class ChatMessageRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    role: str
    text: str
    created_at: datetime


class ChatSessionSummaryRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    session_id: str
    astrologer_id: int
    # Transient — set by chat_log_service, not real columns on ChatSession
    # (same convention as AstrologerRead.priority; see ticket_service).
    astrologer_name: str
    priority: int | None = None
    category: str | None = None
    sub_category: str | None = None
    resolved_by: SessionResolution | None = None
    ticket_id: int | None = None
    started_at: datetime
    resolved_at: datetime | None = None


class ChatSessionDetailRead(ChatSessionSummaryRead):
    messages: list[ChatMessageRead] = []

File: packages/shared/src/types/chatLog.ts
export interface ChatLogMessage {
  role: string;
  text: string;
  created_at: string;
}

export interface ChatSessionSummary {
  id: number;
  session_id: string;
  astrologer_id: number;
  astrologer_name: string;
  priority: number | null;
  category: string | null;
  sub_category: string | null;
  // "bot" | "escalated" | null (still an open, unconcluded conversation)
  resolved_by: string | null;
  ticket_id: number | null;
  started_at: string;
  resolved_at: string | null;
}

export interface ChatSessionDetail extends ChatSessionSummary {
  // Empty for sessions that started before message persistence shipped
  // (2026-08-18) — only ChatSession's own resolution metadata exists for those.
  messages: ChatLogMessage[];
}
